pallets/messages/src/weights.rs

```diff
@@ -35,83 +35,133 @@ use core::marker::PhantomData;
-	/// Proof: `Schemas::SchemaInfos` (`max_values`: None, `max_size`: Some(15), added: 2490, mode: `MaxEncodedLen`)
+	/// Proof: `Schemas::SchemaInfos` (`max_values`: None, `max_size`: Some(17), added: 2492, mode: `MaxEncodedLen`)
 	/// Storage: `Msa::PublicKeyToMsaId` (r:1 w:0)
 	/// Proof: `Msa::PublicKeyToMsaId` (`max_values`: None, `max_size`: Some(48), added: 2523, mode: `MaxEncodedLen`)
 	/// Storage: `Msa::DelegatorAndProviderToDelegation` (r:1 w:0)
 	/// Proof: `Msa::DelegatorAndProviderToDelegation` (`max_values`: None, `max_size`: Some(217), added: 2692, mode: `MaxEncodedLen`)
-	/// Storage: `Messages::MessagesV2` (r:0 w:1)
-	/// Proof: `Messages::MessagesV2` (`max_values`: None, `max_size`: Some(3123), added: 5598, mode: `MaxEncodedLen`)
+	/// Storage: `Messages::MessagesV3` (r:0 w:1)
+	/// Proof: `Messages::MessagesV3` (`max_values`: None, `max_size`: Some(3125), added: 5600, mode: `MaxEncodedLen`)
 	/// The range of component `n` is `[0, 3071]`.
-	fn add_onchain_message(n: u32, ) -> Weight {
+	fn add_onchain_message(_n: u32, ) -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `881`
+		//  Measured:  `548`
 		//  Estimated: `4177`
-		// Minimum execution time: 40_860_000 picoseconds.
-		Weight::from_parts(42_194_321, 4177)
-			// Standard Error: 145
-			.saturating_add(Weight::from_parts(959, 0).saturating_mul(n.into()))
+		// Minimum execution time: 26_000_000 picoseconds.
+		Weight::from_parts(38_413_044, 4177)
 			.saturating_add(T::DbWeight::get().reads(3_u64))
 			.saturating_add(T::DbWeight::get().writes(1_u64))
 	}
 	/// Storage: `Schemas::SchemaInfos` (r:1 w:0)
-	/// Proof: `Schemas::SchemaInfos` (`max_values`: None, `max_size`: Some(15), added: 2490, mode: `MaxEncodedLen`)
+	/// Proof: `Schemas::SchemaInfos` (`max_values`: None, `max_size`: Some(17), added: 2492, mode: `MaxEncodedLen`)
 	/// Storage: `Msa::PublicKeyToMsaId` (r:1 w:0)
 	/// Proof: `Msa::PublicKeyToMsaId` (`max_values`: None, `max_size`: Some(48), added: 2523, mode: `MaxEncodedLen`)
-	/// Storage: `Messages::MessagesV2` (r:0 w:1)
-	/// Proof: `Messages::MessagesV2` (`max_values`: None, `max_size`: Some(3123), added: 5598, mode: `MaxEncodedLen`)
+	/// Storage: `Messages::MessagesV3` (r:0 w:1)
+	/// Proof: `Messages::MessagesV3` (`max_values`: None, `max_size`: Some(3125), added: 5600, mode: `MaxEncodedLen`)
 	fn add_ipfs_message() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `814`
+		//  Measured:  `471`
 		//  Estimated: `4008`
-		// Minimum execution time: 31_646_000 picoseconds.
-		Weight::from_parts(32_704_000, 4008)
+		// Minimum execution time: 25_000_000 picoseconds.
+		Weight::from_parts(43_000_000, 4008)
 			.saturating_add(T::DbWeight::get().reads(2_u64))
 			.saturating_add(T::DbWeight::get().writes(1_u64))
 	}
+	/// Storage: UNKNOWN KEY `0x9ea3e2d10fdb9a071f2f534d51b0961f963740dfb0edb77dddcbb1e5542cf4d2` (r:2 w:1)
+	/// Proof: UNKNOWN KEY `0x9ea3e2d10fdb9a071f2f534d51b0961f963740dfb0edb77dddcbb1e5542cf4d2` (r:2 w:1)
```

Comment:
why is there 2 reads on this? Is it because of the `iter.next()` ? and what we talked about reading prefix keys. Since it is retrieving the key and value in the same call I was going to assume it's just 1 read for reading both and 1 write for drain functionality but this is still weird.